full-source-pretask.yml

```diff
@@ -0,0 +1,56 @@
+---
+- block:
+
+  # Most of packages are required by pulp_rpm
+  # https://pulp-rpm.readthedocs.io/en/latest/installation.html#install-createrepo-c-from-source
+  - name: Install pre-requisites for plugin installations
+    package: name={{ item }} state=present
+    loop:
+      - git
+      - createrepo
+      - createrepo_c
+      - python3-createrepo_c
+      - gcc
+      - make
+      - cmake
+      - bzip2-devel
+      - expat-devel
+      - file-devel
+      - glib2-devel
+      - libcurl-devel
+      - libxml2-devel
+      - python3-devel
+      - rpm-devel
+      - openssl-devel
+      - sqlite-devel
+      - xz-devel
+      - zlib-devel
+
+  become: true
+
+- block:
+
+  # Pip should be able to write to /pulp_source/* directories
+
+  - name: Create source folder
+    file: path=/pulp_source state=directory mode=0777
+
+  - name: Clone pulp source repository
+    git:
+      repo: '{{ pulp_git_repo }}'
+      dest: '{{ pulp_source_dir }}'
+      force: true
+    when: pulp_git_repo is defined
+
+  - name: Clone plugins source repositories
+    git:
+      repo: '{{ item.value.git_repo }}'
+      dest: '{{ item.value.source_dir }}'
+      force: true
+    with_dict: '{{ pulp_install_plugins }}'
+    when: pulp_install_plugins[item.key].git_repo is defined
+
+  - name: set permissions
+    shell: chmod -R 0777 /pulp_source/*
+
+  become: true
```

Comment:
If all of this is required for a source install, should the pulp3 role handle this?

Reply:
@ehelms this is required only if the source is cloned by the playbook, for dev installs there are people using vagrant and then with vagrant it is not needed.